File: www/SocketIO.js
class MediaPlayer extends tabris.NativeObject {
  constructor(properties) {
    super(properties);
  }

  get _nativeType() {
    return 'ru.fax1ty.tabris.socket_io.SocketIO';
  }

  emit(event, ...args) {
    if (typeof event != 'string') throw new Error('Event argument must be a string!');
    else {
      if (!args) this._nativeCall('emit', { event: event });
      else this._nativeCall('emit', { event: event, args: args });
    }
  }

  disconnect() {
    this._nativeCall('disconnect');
  }

  _trigger(name, event) {
    return super._trigger(name, event);
  }

  _listen(name, listening) {
    this._nativeListen(name, listening);
  }
}

tabris.NativeObject.defineProperties(MediaPlayer.prototype, {
  'url': {
    type: 'string',
    default: ''
  }
});

module.exports = MediaPlayer;

File: src/android/ru/fax1ty/tabris/socket_io/SocketIO.kt
package ru.fax1ty.tabris.socket_io

import android.app.Activity
import io.socket.client.IO

class SocketIO(activity: Activity, url: String) {

    private val socket = IO.socket(url)

    fun emit(event: String) {
        socket.emit(event)
    }

    fun emit(event: String, args: List<Any>?) {
        socket.emit(event, args)
    }

    fun on(event: String, listener: OnListener) {
        socket.on(event, listener)
    }

    fun off(event: String) {
        socket.off(event)
    }

    fun disconnect() {
        socket.disconnect()
    }
}